### ISSUE TITLE
PHP 8.1 by reference has space before

### ISSUE THREAD

**Issue by alexander-schranz** [FriendsOfPHP/PHP-CS-Fixer#5993]
## Bug report

> PHP CS Fixer 3.1.0 River by Fabien Potencier and Dariusz Ruminski
> PHP 8.1.0RC2-dev (cli) (built: Sep  2 2021 18:40:02) (NTS)

<details>
<summary>.php-cs-fixer.dist.php</summary>

```
<?php

$header = <<<EOF
This file is part of Sulu.

(c) Sulu GmbH

This source file is subject to the MIT license that is bundled
with this source code in the file LICENSE.
EOF;

$finder = PhpCsFixer\Finder::create()
    ->exclude(['var/cache', 'tests/Resources/cache', 'node_modules'])
    ->in(__DIR__);

$config = new PhpCsFixer\Config();
$config->setRiskyAllowed(true)
    ->setRules([
        '@Symfony' => true,
        'array_syntax' => ['syntax' => 'short'],
        'class_definition' => false,
        'concat_space' => ['spacing' => 'one'],
        'function_declaration' => ['closure_function_spacing' => 'none'],
        'header_comment' => ['header' => $header],
        'native_constant_invocation' => true,
        'native_function_casing' => true,
        'native_function_invocation' => ['include' => ['@internal']],
        'no_superfluous_phpdoc_tags' => ['allow_mixed' => true, 'remove_inheritdoc' => true],
        'ordered_imports' => true,
        'phpdoc_align' => ['align' => 'left'],
        'phpdoc_types_order' => false,
        'single_line_throw' => false,
    ])
    ->setFinder($finder);

return $config;
```

</details>

When running php-cs-fixer on 8.1 by reference is different to php 8.0 or lower. I tested also master and 8.1 PR branch still got the same result.

### Code snippet that reproduces the problem

Using `PHP_CS_FIXER_IGNORE_ENV` as PHP 8.1 is not yet official supported.

```php
PHP_CS_FIXER_IGNORE_ENV=1 vendor/bin/php-cs-fixer fix src/Sulu/Bundle/SecurityBundle/Tests/Unit/Controller/PermissionControllerTest.php -vvv
```

```
src/Sulu/Bundle/SecurityBundle/Tests/Unit/Controller/PermissionControllerTest.php (single_space_after_construct, function_declaration, function_typehint_space)
      ---------- begin diff ----------
--- /home/runner/work/sulu/sulu/src/Sulu/Bundle/SecurityBundle/Tests/Unit/Controller/PermissionControllerTest.php
+++ /home/runner/work/sulu/sulu/src/Sulu/Bundle/SecurityBundle/Tests/Unit/Controller/PermissionControllerTest.php
@@ -138,7 +138,7 @@
 
         \array_walk(
             $permissions,
-            function(&$permissionLine) {
+            function(& $permissionLine) {
                 $permissionLine = 'true' === $permissionLine || true === $permissionLine;
             }
         );
```

On PHP 8.0 or lower it is correctly `function(& $permissionLine) {`

**Comment by GrahamCampbell** [comment]
This is a duplicate of https://github.com/FriendsOfPHP/PHP-CS-Fixer/issues/5891 and should be closed.

**Comment by GrahamCampbell** [comment]
NB Work on fixing this started at https://github.com/FriendsOfPHP/PHP-CS-Fixer/pull/5863.

**Comment by alexander-schranz** [comment]
Not sure if its already implemented but I tried #5863 out, there this issue still exist. Looks its more about adding intersection type and about the reference issue?

**Comment by SpacePossum** [comment]
it comes from the same issue; PHP8.1 tokenizing `&` differently for _both_ reference _and_ intersection type, it is just how they implemented it in PHP8.1 (which is a pain for us here in this project, but we code our way out of it somehow)
